### PR TITLE
Bug Fix: Keploy test runs succesfully with no test cases

### PR DESCRIPTION
**Pull request by Akash-Singh04**
## Related Issue
  -Bug Fix 

Closes: #1270 #1750 

#### Describe the changes you've made

1. Added check to ensure ```keploy test``` informs the user that no test cases were found instead of completing the test successfully.
2.  If not all test cases are empty, ```keploy test``` runs normally and informs user of the empty test cases.
3.  Added check to ensure ```keploy test``` informs user of missing Keploy Folder and exits gracefully.

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, local variables)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Please let us know if any test cases are added

NIL

#### Describe if there is any unusual behaviour of your code(Write `NA` if there isn't)
NIL

## Checklist:

- [x] My code follows the style guidelines of this project.
- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [ ] My changes generate no new warnings.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] New and existing unit tests pass locally with my changes.

## Screenshots (if any)

1. image
2. image
3. image

### CODE REVIEW

**Comment by Akash-Singh04** [comment]
@charankamarapu Let me know if this implementation  is apt!

**Comment by charankamarapu** [comment]
Please update the branch . And also have the many number of test-set then how are you going to approach this . Lets say I have 10 test-sets in which only one is empty.

**Review by charankamarapu — COMMENTED** [review]
there are no file changes

**Comment by Akash-Singh04** [comment]
@charankamarapu After the v2 refactoring, i have added warnings so that the user is warned about missing keploy folder, and if only a single test set is empty, the execution of other test sets continues normally whilst informing the user of the missing test set. Please review the PR and refer to the updated pr description for the screenshots

File: pkg/service/replay/replay.go [hunk 1]
```diff
@@ -111,6 +111,13 @@ func (r *Replayer) Start(ctx context.Context) error {
 		return fmt.Errorf(stopReason)
 	}
 
+	if len(testSetIDs) == 0 {
```
Comment (charankamarapu) [inline pkg/service/replay/replay.go:114]: do this before boot replay function.

**Review by charankamarapu — COMMENTED** [review]
Please address the comments

File: pkg/service/replay/replay.go [hunk 2]
```diff
@@ -246,6 +253,8 @@ func (r *Replayer) RunTestSet(ctx context.Context, testSetID string, testRunID s
 	}
 
 	if len(testCases) == 0 {
+		errMsg := fmt.Sprint("No test cases found for the test set: ", models.HighlightGrayString(testSetID))
```
[the remark this answers is not in the document]
Comment (Akash-Singh04) [reply pkg/service/replay/replay.go:256]: Refactored

File: pkg/service/replay/replay.go (hunk 1 above)
Comment (Akash-Singh04) [reply pkg/service/replay/replay.go:114]: Refactored

**Comment by Akash-Singh04** [comment]
@charankamarapu Your requested changes have been made

File: cli/provider/cmd.go [hunk 3]
```diff
@@ -387,6 +388,14 @@ func (c *CmdConfigurator) ValidateFlags(ctx context.Context, cmd *cobra.Command)
 
 		c.cfg.Path = absPath + "/keploy"
 		if cmd.Name() == "test" {
+			//check if the keploy folder exists
+			if _, err := os.Stat(c.cfg.Path); os.IsNotExist(err) {
+				recordCmd := models.HighlightGrayString("keploy record")
+				errMsg := fmt.Sprintf("Keploy folder not found. Please record testcases using %s command", recordCmd)
+				utils.LogError(c.logger, err, errMsg)
```
Comment (charankamarapu) [inline cli/provider/cmd.go:395]: Hey just saw the screenshot and I think the error shouldn't include stat error and it will be good if error is "no test-sets found" rather than "no keploy folder" because keploy folder has no usecase to user but test-sets does. Please update the screenshots after this change.

**Review by charankamarapu — COMMENTED** [review]
Please address the comments

File: cli/provider/cmd.go (hunk 3 above)
Comment (Akash-Singh04) [reply cli/provider/cmd.go:395]: Refactored and updated screenshot no.3

**Comment by Akash-Singh04** [comment]
@charankamarapu Your requested changes have been made

**Review by charankamarapu — APPROVED** [review]
LGTM
